### PR TITLE
DEV-6326 Allow custom functions to be benchmarked

### PR DESCRIPTION
**Pull request by JosVerburg**
Please make sure to review the compare.benchmark.ts carefully. It's important that the "manual" code does the same as the "xpath" version. I've already confirmed that the end result of both are equal.

### CODE REVIEW

**Comment by mehmetscoskun** [comment]
when I run npm run performance, it will run all performance tests, right? Is it too hard to add something like this: `runner.only.addBenchmark`  or `runner.only.compareBenchmark` . Maybe this one can take time but we can pass an option like this `{only: true/false, skip:true/false}` to `addBenchmark` and `compareBenchmark  functions`. I think it would be useful. at least in the future

File: performance/server.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+					modulesFileServer.serve(request, response);
+					break;
+				default:
+					console.log('Serving ./performance' + request.url);
+					performanceFileServer.serve(request, response);
+					break;
+			}
+		})
+		.resume();
+}).listen(8080);
```
[the remark this answers is not in the document]
Comment (JosVerburg) [reply performance/server.ts:33]: What do you mean to do it together with the other one? Is there a story or an issue to make that one configurable?

[the remark this answers is not in the document]
Comment (JosVerburg) [reply performance/server.ts:33]: Turns out this is not easily doable. https://github.com/kimmobrunfeldt/concurrently/issues/33
I'd prefer to just keep 8080 as static port number for now as is also done in the demo/server.ts you linked.
If you want to have support for this than I suggest that we create a separate issue for this, if you do so please also include the github issue I referenced as the library "concurrently" which we are using now creates the problem.

**Comment by JosVerburg** [comment]
@mehmetscoskun I've added the API to run only 1 performance benchmark. This is the `runner.only.addBenchmark` format. You can check the latest commit for this.

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /coverage
 /node_modules
+/performance/lib
```
[the remark this answers is not in the document]
Comment (JosVerburg) [reply .gitignore:3]: Check it with an `npm build` and `npm pack` could not see signs of the benchmark tooling in either.

**Pull request merged**